src/aws/auth.rs

```diff
@@ -1,6 +1,7 @@
 //! Authentication settings for AWS components.
 use std::time::Duration;
 
+#[allow(deprecated)]
```

Comment:
Why do we need these deprecation allows?

Reply:
We have `#![deny(warnings)]` which implies deny deprecated. Running the license tool seemed to cause some cascading library bumps downstream from the necessary bumps to eg aws-config (https://github.com/vectordotdev/vector/pull/21905/files#diff-13ee4b2252c9e516a0547f2891aa2105c3ca71c6d7a1e682c69be97998dfc87eR779), and the newer version of aws-config has ProfileFileKind and ProfileFiles as deprecated.

However, now that I look, it seems like EnvConfigFileKind and EnvConfigFiles from aws-runtime are drop-in replacements, so I'll go ahead and make those changes.

Reply:
Thanks for resolving those!